src/core/server/events/events.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  CommentCreatedInput,
+  CommentEnteredModerationQueueInput,
+  CommentFeaturedInput,
+  CommentLeftModerationQueueInput,
+  CommentReleasedInput,
+  CommentReplyCreatedInput,
+  CommentStatusUpdatedInput,
+} from "coral-server/graph/resolvers/Subscription";
+
+import { CoralEventPayload, createCoralEvent } from "./event";
+import { CoralEventType } from "./types";
+
+export type CommentEnteredModerationQueueCoralEventPayload = CoralEventPayload<
+  CoralEventType.COMMENT_ENTERED_MODERATION_QUEUE,
+  CommentEnteredModerationQueueInput
+>;
+
+export const CommentEnteredModerationQueueCoralEvent = createCoralEvent<
+  CommentEnteredModerationQueueCoralEventPayload
+>(CoralEventType.COMMENT_ENTERED_MODERATION_QUEUE);
+
+export type CommentLeftModerationQueueCoralEventPayload = CoralEventPayload<
+  CoralEventType.COMMENT_LEFT_MODERATION_QUEUE,
+  CommentLeftModerationQueueInput
+>;
+
+export const CommentLeftModerationQueueCoralEvent = createCoralEvent<
+  CommentLeftModerationQueueCoralEventPayload
+>(CoralEventType.COMMENT_LEFT_MODERATION_QUEUE);
+
+export type CommentStatusUpdatedCoralEventPayload = CoralEventPayload<
+  CoralEventType.COMMENT_STATUS_UPDATED,
+  CommentStatusUpdatedInput
+>;
+
+export const CommentStatusUpdatedCoralEvent = createCoralEvent<
+  CommentStatusUpdatedCoralEventPayload
+>(CoralEventType.COMMENT_STATUS_UPDATED);
+
+export type CommentReplyCreatedCoralEventPayload = CoralEventPayload<
+  CoralEventType.COMMENT_REPLY_CREATED,
+  CommentReplyCreatedInput
+>;
+
+export const CommentReplyCreatedCoralEvent = createCoralEvent<
+  CommentReplyCreatedCoralEventPayload
+>(CoralEventType.COMMENT_REPLY_CREATED);
+
+export type CommentCreatedCoralEventPayload = CoralEventPayload<
+  CoralEventType.COMMENT_CREATED,
+  CommentCreatedInput
+>;
+
+export const CommentCreatedCoralEvent = createCoralEvent<
+  CommentCreatedCoralEventPayload
+>(CoralEventType.COMMENT_CREATED);
+
+export type CommentFeaturedCoralEventPayload = CoralEventPayload<
+  CoralEventType.COMMENT_FEATURED,
+  CommentFeaturedInput
+>;
+
+export const CommentFeaturedCoralEvent = createCoralEvent<
+  CommentFeaturedCoralEventPayload
+>(CoralEventType.COMMENT_FEATURED);
+
+export type CommentReleasedCoralEventPayload = CoralEventPayload<
+  CoralEventType.COMMENT_RELEASED,
+  CommentReleasedInput
+>;
+
+export const CommentReleasedCoralEvent = createCoralEvent<
+  CommentReleasedCoralEventPayload
+>(CoralEventType.COMMENT_RELEASED);
+
+export type StoryCreatedCoralEventPayload = CoralEventPayload<
+  CoralEventType.STORY_CREATED,
+  {
+    storyID: string;
+    storyURL: string;
+  }
+>;
+
+export const StoryCreatedCoralEvent = createCoralEvent<
+  StoryCreatedCoralEventPayload
+>(CoralEventType.STORY_CREATED);
```

Comment:
I think it should be possible to instead of:
```ts
export type CommentReleasedCoralEventPayload = CoralEventPayload<
  CoralEventType.COMMENT_RELEASED,
  CommentReleasedInput
>;

export const CommentReleasedCoralEvent = createCoralEvent<
  CommentReleasedCoralEventPayload
>(CoralEventType.COMMENT_RELEASED);
```
To do:
```ts
export const CommentReleasedCoralEvent = createCoralEvent<
  CommentReleasedInput
>(CoralEventType.COMMENT_RELEASED);
```
and have a Type function that returns the Payload whenever you need:
```ts
type CommentReleasedCoralEvent = CoralEventPayload<typeof CommentReleasedCoralEvent>;
```
But I haven't actually validated it yet.

Reply:
I've looked into this too, but there's not really a clean way to extract the generic type there, and your proposal loses the binding of the data type to the type name, meaning we don't get the free type switching provided by the current system. It's not as pretty, but nothing wrong defining a type like that.

I was actually considering naming them the same:

```ts
export type CommentLeftModerationQueueCoralEventPayload = CoralEventPayload<
  CoralEventType.COMMENT_LEFT_MODERATION_QUEUE,
  CommentLeftModerationQueueInput
>;

export const CommentLeftModerationQueueCoralEvent = createCoralEvent<
  CommentLeftModerationQueueCoralEventPayload
>(CoralEventType.COMMENT_LEFT_MODERATION_QUEUE);
```

becomes:

```ts
export type CommentLeftModerationQueueCoralEvent = CoralEventPayload<
  CoralEventType.COMMENT_LEFT_MODERATION_QUEUE,
  CommentLeftModerationQueueInput
>;

export const CommentLeftModerationQueueCoralEvent = createCoralEvent<
  CommentLeftModerationQueueCoralEvent
>(CoralEventType.COMMENT_LEFT_MODERATION_QUEUE);
```

And Typescript would select the type/value depending on the usage automatically.